bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,507 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.somfymylink.internal.SomfyMyLinkBindingConstants.*;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link SomfyMyLinkBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Chris Johnson - Initial contribution
+ */
+@NonNullByDefault
+public class SomfyMyLinkBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyMyLinkBridgeHandler.class);
+
+    @Nullable
+    private SomfyMyLinkConfiguration config;
+
+    private static final int HEARTBEAT_MINUTES = 2;
+    private static final int MYLINK_PORT = 44100;
+    private static final int MYLINK_DEFAULT_TIMEOUT = 5000;
+    private static final Object CONNECTION_LOCK = new Object();
+    private static final int CONNECTION_DELAY = 1000;
+
+    private static final String MYLINK_COMMAND_TEMPLATE = "{\"id\": %1$s, \"method\": \"%2$s\",\"params\": {\"%3$s\": %4$s,\"auth\": \"%5$s\"}}";
+
+    @Nullable
+    private ScheduledFuture<?> heartbeat;
+
+    @Nullable
+    private ServiceRegistration<DiscoveryService> discoveryServiceRegistration;
+
+    private SomfyMyLinkDeviceDiscoveryService discovery;
+
+    @Nullable
+    private SomfyMyLinkStateDescriptionOptionsProvider stateDescriptionProvider;
+
+    // Gson & parser
+    private final Gson gson = new Gson();
+
+    public SomfyMyLinkBridgeHandler(Bridge bridge, @Nullable SomfyMyLinkStateDescriptionOptionsProvider stateDescriptionProvider) {
+        super(bridge);
+
+        this.discovery = new SomfyMyLinkDeviceDiscoveryService(this);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command received on mylink {}", command);
+
+        try {
+            if (CHANNEL_SCENES.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    return;
+                }
+
+                if (CHANNEL_SCENES.equals(channelUID.getId()) && command instanceof StringType) {
+                    Integer sceneId = Integer.decode(command.toString());
+                    commandScene(sceneId);
+                }
+            }
+        } catch (SomfyMyLinkException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing mylink");
+        config = getThing().getConfiguration().as(SomfyMyLinkConfiguration.class);
+
+        if (validConfiguration(config)) {
+            this.discoveryServiceRegistration = this.bundleContext.registerService(DiscoveryService.class, this.discovery, null);
```

Comment:
As `bundleContext` has been removed from `BaseThingHandler`, this needs to be changed to use `FrameworkUtil.getBundle(SomfyMyLinkBridgeHandler.class).getBundleContext()`.

Reply:
done